mingw-w64-git/PKGBUILD

```
@@ -0,0 +1,141 @@
+# Maintainer: Yuyi Wang <Strawberry_Str@hotmail.com>
+
+_realname=git
+pkgbase="mingw-w64-${_realname}"
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=2.39.2.windows.1
+pkgrel=1
+pkgdesc="The fast distributed version control system (mingw-w64)"
+arch=('any')
+url="https://git-for-windows.github.io/"
+license=('GPL2')
+groups=('VCS')
+provides=('git' 'git-core')
```

```suggestion
provides=('git' 'git-core')
```
I don't think mingw-w64 could provide an msys2 one.

---

Then I need to change several other packages because `git-lfs`, `delta` etc. now depends on MSYS2 git, but they could depend on MinGW git.

---

> Then I need to change several other packages because `git-lfs`, `delta` etc. now depends on MSYS2 git, but they could depend on MinGW git.

The main problem is that many packages depend on `git` as `makedepends`, so you may need to change all of then:(

---

Well, not all. Those packages that use `git` as a make dependency, should still use `git` because `makepkg` is a bash script.

---

In theory there is no reason why they couldn't switch to mingw git.

---

It's the maintainers' turn to choose which git to use. I don't need to bother them in this PR.

---

> It's the maintainers' turn to choose which git to use. I don't need to bother them in this PR.

It seems that msys2 does not have a complex and hierarchical team structure like Arch Linux (which has core team and TU), so the dependencies of most packages are determined by the official.

---

OK then it's official's turn:) I don't think it is easy to switch msys2 git to mingw git as a make dep. It needs support from makepkg. I would rather not changing them. What would I like to change is the git tools that depends on git only because a user cannot use them directly. A mingw git will work well without large change.

---

Yes, that can all wait..

---

Yes, I now change to only provide `git-core`. Then it won't break the existing scripts and worktrees.

---

But you cause confusion because it is not at all clear why `git` _and_ `git-core` exist.